controller/src/main/java/io/pravega/controller/store/stream/Segment.java

```diff
@@ -20,12 +20,11 @@
 @Data
 @ToString(includeFieldNames = true)
 public class Segment {
-
-    protected final int number;
-    protected final int epoch;
-    protected final long start;
-    protected final double keyStart;
-    protected final double keyEnd;
+    private final long segmentId;
```

Comment:
Should this be split here? Or should we have getters for the primary at least?

Reply:
The primary and secondary extraction is implemented in StreamSegmentNameUtils. 
All business logic that needs to extract primary or secondary uses that utility class to do so.

Reply:
yes its redundant. I can simply have a getter which extracts epoch from segment id